File: scripts/user/RepoInfo.java
//package com.repo.info;

// Update by scripts/sync-repo-info
public class RepoInfo {
    // build host name, user name, os name version
    static public final String hostName = "host name";
    static public final String hostUser = "user name";
    static public final String hostOsNV = "Ubuntu 14.04";

    // build usr, emai, data time
    static public final String buildUser = "user-name <email@demo.com>";
    static public final String buildTime = "2018-01-30 10:20:30 +0845";

    // source modify date time
    static public final String modifyTime = "2019-01-30 20:50:59 +8123";

    // remote repo url & hash
    static public final String repoHash = "615"; // SVN Reversion or Git Commit Hash
    static public final String repoUrl = "svn://addr/app/trunk/mta"; // remote repo url

    // semantic version
    static public final long major = 11;
    static public final long minor = 22;
    static public final long patch = 33;
    static public final String tweak = "alpha";
    static public final String semver = "11.22.33-alpha";

    public static void main(String[] args) {
        System.out.println(hostName);
        System.out.println(hostUser);
        System.out.println(hostOsNV);

        System.out.println(buildUser);
        System.out.println(buildTime);

        System.out.println(modifyTime);
        System.out.println(repoHash);
        System.out.println(repoUrl);

        System.out.println(major);
        System.out.println(minor);
        System.out.println(patch);
        System.out.println(tweak);
        System.out.println(semver);
    }
}

File: scripts/test/user/RepoInfo.cpp
#include <string>
#include <iostream>

// Update by scripts/sync-repo-info
struct RepoInfo {
    // build host name, user name, os name version
    static const std::string hostName;
    static const std::string hostUser;
    static const std::string hostOsNV;

    // build usr, emai, data time
    static const std::string buildUser;
    static const std::string buildTime;

    // source modify date time
    static const std::string modifyTime;

    // remote repo url & hash
    static const std::string repoHash; // SVN Reversion or Git Commit Hash
    static const std::string repoUrl; // remote repo url

    // semantic version
    static const std::string major;
    static const std::string minor;
    static const std::string patch;
    static const std::string tweak;
    static const std::string semver;
};

const std::string RepoInfo::hostName = "host name";
const std::string RepoInfo::hostUser = "user name";
const std::string RepoInfo::hostOsNV = "Ubuntu 14.04";

const std::string RepoInfo::buildUser = "user-name <email@demo.com>";
const std::string RepoInfo::buildTime = "2018-01-30 10:20:30 +0845";

const std::string RepoInfo::modifyTime = "2019-01-30 20:50:59 +8123";

const std::string RepoInfo::repoHash = "615";
const std::string RepoInfo::repoUrl = "svn://addr/app/trunk/mta";

const std::string RepoInfo::major = "11";
const std::string RepoInfo::minor = "22";
const std::string RepoInfo::patch = "33";
const std::string RepoInfo::tweak = "rc.20190423";
const std::string RepoInfo::semver = "11.22.33-rc.20190423";

int main(void)
{
    std::cout << RepoInfo::hostName << std::endl;
    std::cout << RepoInfo::hostUser << std::endl;
    std::cout << RepoInfo::hostOsNV << std::endl;

    std::cout << RepoInfo::buildUser << std::endl;
    std::cout << RepoInfo::buildTime << std::endl;

    std::cout << RepoInfo::modifyTime << std::endl;
    std::cout << RepoInfo::repoHash << std::endl;
    std::cout << RepoInfo::repoUrl << std::endl;

    std::cout << RepoInfo::major << std::endl;
    std::cout << RepoInfo::minor << std::endl;
    std::cout << RepoInfo::patch << std::endl;
    std::cout << RepoInfo::tweak << std::endl;

    return 0;
}
